splinepy/_spline.py

```diff
@@ -566,56 +671,71 @@ def _check_and_update_c(self,):
         -------
         None
         """
+        required_props = self.required_properties
+
         # Check if all data is available for spline type
-        for i_property in self._required_properties:
+        for i_property in required_props:
             if getattr(self, i_property) is None:
                 logging.debug(
                     "Spline - Not enough information to update cpp spline. "
                     "Skipping update."
                 )
-                return
+                return None
 
         # Check if data is coherent
         # Check if control point dimension matches dim dimension
-        if not self.dim == self.control_points.shape[1]:
-            raise InputDimensionError(
-                "Mismatch between dimension and control point dimensionality"
-            )
+        #if not self.dim == self.control_points.shape[1]:
+        #    raise InputDimensionError(
+        #        "Mismatch between dimension and control point dimensionality"
+        #    )
         # Check if given degree match expected number of degrees
-        if not self.degrees.shape[0] == self.para_dim:
-            raise InputDimensionError(
-                "RDimension mismatch between degrees and parametric dimension"
-            )
+        #if not self.degrees.shape[0] == self.para_dim:
+        #    raise InputDimensionError(
+        #        "RDimension mismatch between degrees and parametric dimension"
+        #    )
+
```

Comment:
It seemed a bit redundant, but if this was to fight against possible in-place manipulation, we can keep it.

Reply:
++ I will remove comments before merging